dvc/scm.py

```diff
@@ -47,6 +49,25 @@ def __init__(self, reason: str) -> None:
         super().__init__(f"{reason}\n{doc}")
 
 
+class GitMergeError(SCMError):
+    def __init__(self, msg: str, scm: Optional["Git"] = None) -> None:
+        if scm and self._is_shallow(scm):
+            url = format_link(
+                "https://dvc.org/doc/user-guide/troubleshooting#git-shallow"
```

Comment:
depends on https://github.com/iterative/dvc.org/pull/3577

Reply:
Also you can use `error_link("git-shallow")` instead of `format_link`.